unittests/protocol_feature_tests.cpp

```diff
@@ -628,14 +628,10 @@ BOOST_AUTO_TEST_CASE(steal_my_ram) {
       );
 
       wlog("Moving data around with bob's authorization...");
-      BOOST_REQUIRE_EXCEPTION(
-         c.push_action(tester1_account, "setdata"_n, bob_account, mutable_variant_object()
-            ("len1", 0)
-            ("len2", 10)
-            ("payer", alice_account)
-         ),
-         unsatisfied_authorization,
-         fc_exception_message_is("Requested payer alice did not authorize payment")
+      c.push_action(tester1_account, "setdata"_n, bob_account, mutable_variant_object()
+         ("len1", 0)
+         ("len2", 10)
+         ("payer", alice_account)
```

Comment:
I'd love to see an attempt to move from `len1 10 len2 0` to `len1 0 len2 20` fail before we do the successful net-0 transfer.  Also, would be nice to reverse the transfer intentionally losing ram (ie, to `len1 5 len2 10`) and verify that the ram is allocated/refunded appropriately. (if that's done below and I missed it, feel free to disregard)